tests/countries/test_saint_helena_ascension_and_tristan_da_cunha.py

```diff
@@ -0,0 +1,248 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import GOVERNMENT
+from holidays.countries.saint_helena_ascension_and_tristan_da_cunha import (
+    SaintHelenaAscensionAndTristanDaCunha,
+    SH,
+    SHN,
+)
+from tests.common import CommonCountryTests
+
+
+class TestSaintHelenaAscensionAndTristanDaCunha(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(2015, 2050)
+        super().setUpClass(
+            SaintHelenaAscensionAndTristanDaCunha, years=years, years_non_observed=years
+        )
+        cls.government_holidays = SaintHelenaAscensionAndTristanDaCunha(
+            categories=GOVERNMENT, years=years
+        )
+
+    def test_country_aliases(self):
+        self.assertAliases(SaintHelenaAscensionAndTristanDaCunha, SH, SHN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(SaintHelenaAscensionAndTristanDaCunha(years=2014))
+
+    def test_special_holidays(self):
```

Comment:
If you want to use a new approach to this test, it's recommended to do it like in Gibraltar, for example.